core/io/resource_loader.cpp

```diff
@@ -277,6 +277,11 @@ RES ResourceLoader::_load(const String &p_path, const String &p_original_path, c
 
 	ERR_FAIL_COND_V_MSG(found, RES(), "Failed loading resource: " + p_path + ".");
 
+#ifdef TOOLS_ENABLED
+	FileAccessRef file_check = FileAccess::create(FileAccess::ACCESS_RESOURCES);
+	ERR_FAIL_COND_V_MSG(!file_check->file_exists(p_path), RES(), "Resource file not found: " + p_path + ".");
+#endif
```

Comment:
Shouldn't this come before the `found` error?
IIUC, `found` will be true if a valid loader was found for the given path (based on the extension), so if you try to load `res://doesnt_exist.png`, I guess it will show the first error and not the one about the file actually not existing.

Reply:
Nope:
```
_load: No loader found for resource: res://test.png.
```
The first error is for a case when the file exists, but loader failed to load it (e.g. corrupted image).